backend/auth.go

```diff
@@ -0,0 +1,57 @@
+package backend
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+
+	bpb "github.com/Bit-Nation/protobuffers"
+)
+
+// authentication request handler
+func (b *Backend) auth(req *bpb.BackendMessage_Request) (*bpb.BackendMessage_Response, error) {
+
+	auth := req.Auth
+
+	if auth != nil {
+
+		if len(auth.ToSign) != 4 {
+			return nil, errors.New("got invalid amount of toSign bytes")
+		}
+
+		myBytes := make([]byte, 4)
+		_, err := rand.Read(myBytes)
+		if err != nil {
+			return nil, err
+		}
+
+		toSign := append(myBytes, auth.ToSign...)
```

Comment:
Instinctively, I would switch the order here, as usually then we append something we add something to the right side of the original data. In this way the backend can use bytes.HasPrefix()

`toSign := append(auth.ToSign, myBytes...)`

Reply:
Make sense. I switched it